src/@types/global.d.ts

```diff
@@ -37,4 +37,17 @@ declare global {
     interface StorageEstimate {
         usageDetails?: {[key: string]: number};
     }
+
+    export interface ISettledFulfilled<T> {
+        status: "fulfilled";
+        value: T;
+    }
+    export interface ISettledRejected {
+        status: "rejected";
+        reason: any;
+    }
+
+    interface PromiseConstructor {
+        allSettled<T>(promises: Promise<T>[]): Promise<Array<ISettledFulfilled<T> | ISettledRejected>>;
+    }
```

Comment:
It's a little concerning we are defining Promise types ourselves, but this is probably fine for now

Reply:
its an extension

Reply:
that's almost worse :(

Reply:
Its because https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled is still being implemented, we just use it if the browser already supports it else we use a fallback.

Reply:
ah, fair enough.